ops/lib.rs

```diff
@@ -0,0 +1,127 @@
+use proc_macro::TokenStream;
+use quote::quote;
+
+#[proc_macro_attribute]
+pub fn op(_attr: TokenStream, item: TokenStream) -> TokenStream {
+  let func = syn::parse::<syn::ItemFn>(item).expect("expected a function");
+  let name = &func.sig.ident;
+  let generics = &func.sig.generics;
+  let type_params = &func.sig.generics.params;
+  let where_clause = &func.sig.generics.where_clause;
+
+  let inputs = &func.sig.inputs;
+  let output = &func.sig.output;
+  let a = match &inputs[1] {
+    syn::FnArg::Typed(pat) => match *pat.pat {
+      syn::Pat::Wild(_) => quote! {
+        let a = ();
+      },
+      _ => quote! {
+        let a = args.get(2);
+        let a = deno_core::serde_v8::from_v8(scope, a).unwrap();
+      },
+    },
+    _ => unreachable!(),
+  };
+
+  let b = match &inputs[2] {
```

Comment:
Is it possible here to allow for ops with 1 or 2 function arguments instead of 3?

Reply:
just to beautify the syntax a bit (dropping the now redundant _: () arguments)

Reply:
Also, some ops don't even require an op_state; not sure if rust would be smart enough to not borrow state_rc, so allowing an ops signature and not having no such argument might even improve performance.

Reply:
> Is it possible here to allow for ops with 1 or 2 function arguments instead of 3?

The first argument is the OpId used for tracking purposes

Reply:
> Also, some ops don't even require an op_state; not sure if rust would be smart enough to not borrow state_rc, so allowing an ops signature and not having no such argument might even improve performance.

Ops might not but metrics tracking does require borrowing state.

Reply:
> > Is it possible here to allow for ops with 1 or 2 function arguments instead of 3?
> 
> The first argument is the OpId used for tracking purposes

Not sure if I was clear enough here, I meant to say that it is now possible to also allow ops definitions with a function signature that has less than 3 arguments:

eg. https://github.com/denoland/deno/pull/13861/files#diff-fa76844bfec3519648a2d7493933ff3317fa3250dd913cc26d0e86b3b5d4465aR184
```rust
#[op]
fn op_load(
    state: &mut OpState,
    args: LoadArgs,
) -> Result<Value, AnyError> {
    ...
}
```
instead of:
```rust
#[op]
fn op_load(
    state: &mut OpState,
    args: LoadArgs,
    _: (),
) -> Result<Value, AnyError> {
    ...
}
```

Reply:
thanks for explaining, cool PR!